vllm/compilation/sequence_parallelism.py

```diff
@@ -178,57 +442,238 @@ def get_inputs(self):
-                input=all_reduce,
-                residual=residual,
-                weight=rms_norm_weights,
-                epsilon=self.epsilon,
-            )
+        def replacement(
+            result: torch.Tensor,
+            residual: torch.Tensor,
+            mm_1: torch.Tensor,
+            rms_norm_weights: torch.Tensor,
+            scale: torch.Tensor,
+        ) -> tuple[torch.Tensor, torch.Tensor]:
+            reduce_scatter = self._reduce_scatter(mm_1)
+            rs_result = torch.empty_like(reduce_scatter, dtype=result.dtype)
+            rmsnorm = self._functional_fused_add_rmsnorm_quant(
+                rs_result, reduce_scatter, residual, rms_norm_weights, scale)
+            normalized = self._all_gather(rmsnorm[1])
+            return normalized
 
-            return rmsnorm[1]
+        pm.register_replacement(pattern, replacement, self.get_inputs(),
+                                pm.fwd_only, pm_pass)
+
+
+class EmbeddingAllReduceRMSNormStaticFP8Pattern(_RMSNormQuantOpHelper):
+
+    def __init__(self, epsilon: float, dtype: torch.dtype, device: str,
+                 op: torch._ops.OpOverload):
+        super().__init__(epsilon, dtype, device, quant_op=op)
+
+    def get_inputs(self):
+        arg2_1 = torch.empty([16, 4], device=self.device, dtype=self.dtype)
+        mul_6 = torch.tensor([[3, 7, 1, 4, 9, 2, 5, 0]],
+                             device=self.device,
+                             dtype=torch.long)
+        unsqueeze = torch.rand([1, 8, 1], device=self.device,
+                               dtype=self.dtype) > 0.5
+        full_default = torch.zeros([1, 8, 4],
+                                   device=self.device,
+                                   dtype=self.dtype)
+        rmsnorm_result = torch.empty([1, 8, 4],
+                                     device=self.device,
+                                     dtype=self.dtype)
+        quant_result = torch.empty([1, 8, 4],
+                                   device=self.device,
+                                   dtype=FP8_DTYPE)
+        weight = torch.empty([4], device=self.device, dtype=self.dtype)
+        scale = torch.tensor(1.0, device=self.device, dtype=torch.float32)
+        return [
+            arg2_1, mul_6, unsqueeze, full_default, rmsnorm_result,
+            quant_result, weight, scale
+        ]
+
+    def register(self, pm_pass: PatternMatcherPass):
+
+        def pattern(
+            arg2_1: torch.Tensor,
+            mul_6: torch.Tensor,
+            unsqueeze: torch.Tensor,
+            full_default: torch.Tensor,
+            rmsnorm_result: torch.Tensor,
+            quant_result: torch.Tensor,
+            weight: torch.Tensor,
+            scale: torch.Tensor,
+        ):
+            embedding = torch.ops.aten.embedding.default(arg2_1, mul_6)
+            where = torch.ops.aten.where.self(unsqueeze, full_default,
+                                              embedding)
```

Comment:
Is there a reason this `where` cannot be the input and we need the embedding part?